modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts

```diff
@@ -0,0 +1,305 @@
+
+        const multicall = new Multicaller(networkConfig.multicall, jsonRpcProvider, ReliquaryAbi);
+
+        // for the transfer events, we know which users are affected
+        const affetedUsers = transferEvents.flatMap((event) => [event.args.from, event.args.to]);
+        // for the other events, we need to find the owners of the affected relicIds
+        const affectedRelicIds = [
+            ...balanceChangedEvents.map((event) => event.args.relicId),
+            relicManagementEvents.flatMap((event) => [event.args.fromId, event.args.toId]),
+        ];
+
+        affectedRelicIds.forEach((relicId, index) => {
+            multicall.call(`users[${index}]`, reliquaryAddress, 'ownerOf', [relicId]);
+        });
+        let ownerResult: { users: string[] } = await multicall.execute();
+        affetedUsers.push(...ownerResult.users);
+
+        affetedUsers.forEach((userAddress) => {
+            multicall.call(userAddress, reliquaryAddress, 'relicPositionsOfOwner', [userAddress]);
+        });
+
+        // we get a tuple with an array of relicIds and the corresponding positions array
+        const updatedPositions: { [userAddress: string]: [BigNumber[], ReliquaryPosition[]] } =
+            await multicall.execute();
+        // for each user we have to sum up all balances of a specific farm, so we key on user + farmId
+        const userFarmBalances: {
+            [userFarm: string]: { userAddress: string; farmId: string; amount: BigNumber };
+        } = {};
+
+        // we only care for the user address and all positions, we can ignore the relicIds array
+        Object.entries(updatedPositions).forEach(([userAddress, [relicIds, positions]]) => {
+            positions.forEach((position) => {
+                const key = `${userAddress}-${position.poolId}`;
+                if (key in userFarmBalances) {
+                    userFarmBalances[key].amount = userFarmBalances[key].amount.add(position.amount);
+                } else {
+                    userFarmBalances[key] = {
+                        userAddress,
+                        farmId: position.poolId.toString(),
+                        amount: position.amount,
+                    };
+                }
+            });
+        });
+        return Object.values(userFarmBalances).map((userFarmBalance) => ({
+            ...userFarmBalance,
+            amount: formatFixed(userFarmBalance.amount, 18),
+        }));
+    }
+
+    private async loadAllSubgraphRelics(): Promise<ReliquaryRelicFragment[]> {
+        const pageSize = 1000;
+        const MAX_SKIP = 5000;
+        let allRelics: ReliquaryRelicFragment[] = [];
+        let timestamp = 0;
+        let hasMore = true;
+        let skip = 0;
+
+        while (hasMore) {
+            const { relics } = await reliquaryService.getRelics({
+                where: { entryTimestamp_gte: timestamp, balance_gt: '0' },
```

Comment:
shouldn't this be `entryTimestamp_gt` and not `gte` ? Like this you'll get duplicates for the timestamp since they were already in the last batch.

Reply:
using relicId